lib/superagent-mock.js

```diff
@@ -47,6 +47,45 @@ function mock (superagent, config, logger) {
   }();
 
   /**
+   * Chains progrss events and eventually returns the response
+   * @param ProgressEvent (object/function) the ProgressEvent implementation
+   * @param request (object) the current request
+   *
+   * @param progress (object) the progress configuration
+   * @param progress.parts (number)
+   * @param [progress.delay=0] (number)
+   * @param [progress.total=100] (number)
+   * @param [progress.lengthComputable=true] (boolean)
+   * @param [progress.direction='upload'] (string)
+   *
+   * @param remainingParts (number) the remaining calls to make
+   * @param callback (function) the end response callback
+   */
+  var handleProgress = function (ProgressEvent, request, progress, remainingParts, callback) {
+    setTimeout(function () {
+      var total = progress.total || 100;
+      var loaded = total * (progress.parts - remainingParts) / progress.parts;
+      var lengthComputable = progress.lengthComputable !== false; // default is true
+      var e = ProgressEvent.initProgressEvent
+        ? ProgressEvent.initProgressEvent('', true, false, lengthComputable, loaded, total) // IE10+
```

Comment:
I don't like multiline ternary if statement 😷  Is it possible to replace it with a simple `if` ? 😄